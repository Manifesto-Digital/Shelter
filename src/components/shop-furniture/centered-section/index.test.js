import React from 'react';
import { shallow } from 'enzyme';
import { snapshotComponent } from 'test-helpers';
import {
  createFactory,
  createChildContentfulRichText,
} from '../../../utils/test-factories';
import CenteredSection from '.';

// Default props
export const createCenteredSection = createFactory({
  header: 'Mock header',
  text: createChildContentfulRichText(),
  contactNumber: '012345 67890',
});

it('renders correctly', () => {
  const mockData = createCenteredSection();
  snapshotComponent(<CenteredSection {...mockData} />);
});

it('renders a header 2', () => {
  const mockData = createCenteredSection({ header: 'New header text' });
  const wrapper = shallow(<CenteredSection {...mockData} />);
  expect(wrapper.find('h2').text()).toEqual(mockData.header);
});

it('renders the contact number', () => {
  const mockData = createCenteredSection({ contactNumber: '12345 678910' });
  const wrapper = shallow(<CenteredSection {...mockData} />);
  expect(wrapper.text()).toContain(mockData.contactNumber);
});
