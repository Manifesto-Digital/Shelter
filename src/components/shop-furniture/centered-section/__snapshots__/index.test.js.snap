// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  padding: 60px 0;
  background-color: #e5e5e5;
  margin-bottom: 60px;
}

.c2 {
  text-align: center;
  margin: 0 auto;
}

.c2 > p:last-of-type {
  margin-bottom: 0;
}

.c3 {
  color: #ea232d;
}

.c1 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
}

@media (min-width:40.0625em) {
  .c2 {
    max-width: 66.666%;
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h2>
        Mock header
      </h2>
      <div
        className=""
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Our advice and support services across the UK give peopleÂ <b>one-to-one</p>",
          }
        }
      />
      <p>
        Call us on: 
        <span
          className="c3"
        >
          012345 67890
        </span>
      </p>
    </div>
  </div>
</section>
`;
